integration-tests/search.test.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { AxiosInstance } from 'axios';
+import waitForExpect from 'wait-for-expect';
+import { getFhirClient, randomPatient } from './utils';
+
+jest.setTimeout(60 * 1000);
+
+const expectResourceToBePartOfSearchResults = async (
+    client: AxiosInstance,
+    search: { url: string; params?: any },
+    resource: any,
+) => {
+    console.log('Searching with params:', search);
+    await expect(
+        (async () => {
+            return (
+                await client.get(search.url, {
+                    params: search.params,
+                })
+            ).data;
+        })(),
+    ).resolves.toMatchObject({
+        resourceType: 'Bundle',
+        entry: expect.arrayContaining([
+            expect.objectContaining({
+                resource,
+            }),
+        ]),
+    });
+};
+
+describe('search', () => {
+    let client: AxiosInstance;
+    beforeAll(async () => {
+        client = await getFhirClient();
+    });
+    test('search for various valid parameters', async () => {
+        const testPatient: ReturnType<typeof randomPatient> = (await client.post('Patient', randomPatient())).data;
+
+        // wait for the patient to be asynchronously written to ES
+        await waitForExpect(
+            expectResourceToBePartOfSearchResults.bind(
```

Comment:
what does `bind` do here?

Reply:
I'm using it here since `waitForExpect` expects a function as a parameter and I'm "setting" the arguments for expectResourceToBePartOfSearchResults.

If you are not familiar with `.bind` I suggest you read the docs. It is a very powerful/useful function.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind